packages/storage/amplify_storage_s3_dart/lib/src/storage_s3_service/service/storage_s3_service_impl.dart

```diff
@@ -99,7 +96,7 @@ class StorageS3Service {
     final resolvedPrefix = await getResolvedPrefix(
       prefixResolver: _prefixResolver,
       logger: _logger,
-      accessLevel: options.accessLevel,
+      accessLevel: options.accessLevel ?? _s3PluginConfig.defaultAccessLevel,
```

Comment:
I understand this `??` `if null` check is due to `Storage<Operation>Options.accessLevel` is nullable now, and its default value should be set by the plugin, but this default value has been set [here](https://github.com/aws-amplify/amplify-flutter/pull/2793/files#diff-3b6ec49d2366c0c7278efaa1cfc1c8f3d9b845c1f4fcca468113e3bb15cf9b08R172-R175) in the StorageS3Dart implementation (where I think it's a proper location). 

Hum we can't find a better way than `accessLevel: options.accessLevel!` we probably need to remove the default value setting from the StorageS3Dart implementation.

Reply:
using the default value here instead of `accessLevel: options.accessLevel!` makes the code more robust as we do not rely on callers to set value for nullable field.

Reply:
I agree. The storage options interface is not appropriate since it has a nullable access level. Either passing a non-null access level as an additional parameter to the service (e.g. `resolvedAccessLevel`) or handling it inline like this seems fine.

Since other S3 plugin options logic lives here, I think the current handling is appropriate.

Reply:
@dnys1 thanks for suggesting the `resolvedAccessLevel`. I'm more toward handling it inline as there is no logic than checking for null.